Modules/TPC/src/TrendingTaskTPC.cxx

```diff
@@ -0,0 +1,292 @@
+      auto mo = qcdb.retrieveMO(dataSource.path, dataSource.name, timestamp);
+      TObject* obj = mo ? mo->getObject() : nullptr;
+      if (obj) {
+        mReductors[dataSource.name]->update(obj, mSources[dataSource.name],
+                                            dataSource.axisDivision);
+
+        // Number of pads depending if input canvas or single sliced histogram.
+        if (auto canvas = dynamic_cast<TCanvas*>(obj)) { // Case with input canvas.
+          mNumberPads = static_cast<TList*>(canvas->GetListOfPrimitives())->GetEntries();
+        } else if ((int)dataSource.axisDivision[0].size() > 1) { // Case with sliced single histogram.
+          int axisSize = (int)dataSource.axisDivision.size();
+          int innerAxisSize = (int)dataSource.axisDivision[0].size() - 1;
+          mNumberPads = axisSize * innerAxisSize;
+        } else {
+          mNumberPads = 1;
+        }
+      }
+
+    } else if (dataSource.type == "repository-quality") {
+      if (auto qo = qcdb.retrieveQO(dataSource.path + "/" + dataSource.name, timestamp)) {
+        mReductors[dataSource.name]->update(qo.get(), mSources[dataSource.name],
+                                            dataSource.axisDivision);
+      }
+    } else {
+      ILOG(Error, Support) << "Data source '" << dataSource.type << "' unknown." << ENDM;
+    }
+  }
+
+  mTrend->Fill();
+  ILOG(Info, Support) << "'mTrend' has been filled with the current entry." << ENDM;
+}
+
+void TrendingTaskTPC::generatePlots()
+{
+  if (mTrend->GetEntries() < 1) {
+    ILOG(Info, Support) << "No entries in the trend so far, no plot generated." << ENDM;
+    return;
+  }
+
+  ILOG(Info, Support) << "Generating " << mConfig.plots.size() << " plots." << ENDM;
+  for (const auto& plot : mConfig.plots) {
+    // Delete the existing plots before regenerating them.
+    if (mPlots.count(plot.name)) {
+      getObjectsManager()->stopPublishing(plot.name);
+      delete mPlots[plot.name];
+    }
+
+    // Draw the trending on a new canvas.
+    TCanvas* c = new TCanvas();
+    c->SetName(plot.name.c_str());
+    c->SetTitle(plot.title.c_str());
+    drawCanvas(c, plot.varexp, plot.selection, plot.option, plot.graphErrors, plot.name);
+
+    // Postprocess each pad (titles, axes, flushing buffers).
+    for (int p = 1; p <= mNumberPads; p++) {
+      c->cd(p);
+      if (auto histo = dynamic_cast<TGraph*>(c->cd(p)->GetPrimitive("Graph"))) {
+        /*
+                // Set a centered, nicely formatted title.
+                ///if (auto title = static_cast<TPaveText*>(c->GetPrimitive("title"))) {
+                if (auto title = dynamic_cast<TPaveText*>(histo->GetTitle())) {
+                  title->SetBBoxCenterX(c->GetBBoxCenter().fX);
+                  title->Draw();
+                } else {
+                  ILOG(Error, Devel) << "Could not get the title 'TPaveText' of the plot '"
+                    << plot.name << "'." << ENDM;
+                }*/
+        /// LOKI
```

Comment:
As we do not use this part, but is replaced by the nice titles, please remove it